### PR TITLE
salt: Generate custom kubeconfig for Salt Master

### PR DESCRIPTION
**Pull request by gdemonet**
Instead of using the `/etc/kubernetes/admin.conf` file which points to a
specific master's `kube-apiserver` instance, we generate another one
dedicated to Salt Master, and configure it to point to the local
`apiserver-proxy` (which can then route to other masters if the local
one isn't available).

In addition, this kubeconfig generates its own certificate, which could
later map to another group (and thus, other (Cluster)Role(s) than the
current "system:masters").

Note that we reduce the unneeded `/etc/kubernetes` mount to
`/etc/kubernetes/pki` (for SA signing key and etcd encryption key) in
both Salt Master and Salt API containers (SaltAPI didn't need it since
41ba749).

Fixes: #2533

### CODE REVIEW

**Comment by gdemonet** [comment]
This doesn't work with upgrade: obviously, haven't run the `metalk8s.salt.master.kubeconfig` formula yet...

Trying a refactor, moving this PR back to draft state.

**Review by TeddyAndrieux — DISMISSED** [review]

**Review by TeddyAndrieux — DISMISSED** [review]

File: salt/_pillar/metalk8s_nodes.py [hunk 1]
```diff
@@ -36,39 +26,43 @@ def node_info(node, ca_minion):
 
     roles = set()
 
-    if VERSION_LABEL in node.metadata.labels:
-        result['version'] = node.metadata.labels[VERSION_LABEL]
+    if VERSION_LABEL in node['metadata']['labels']:
+        result['version'] = node['metadata']['labels'][VERSION_LABEL]
 
-    for (label, value) in node.metadata.labels.items():
+    for (label, _) in node['metadata']['labels'].items():
```
Comment (gdemonet) [inline salt/_pillar/metalk8s_nodes.py]: `.keys()`?

File: salt/_pillar/metalk8s_nodes.py [hunk 2]
```diff
@@ -36,39 +26,43 @@ def node_info(node, ca_minion):
...
     try:
-        namespace = coreV1.read_namespace(name="kube-system")
-    except (ApiException, HTTPError) as exc:
+        namespace = __salt__['metalk8s_kubernetes.get_object'](
+            name="kube-system",
+            kind="Namespace",
+            apiVersion="v1",
+            kubeconfig=kubeconfig
+        )
+    except Exception as exc:  # pylint: disable=broad-except
```
Comment (gdemonet) [inline salt/_pillar/metalk8s_nodes.py]: It's an execution module, so I guess it should only raise a `CommandExecutionError`?

File: salt/_pillar/metalk8s_nodes.py [hunk 3]
```diff
@@ -156,21 +144,28 @@ def ext_pillar(minion_id, pillar, kubeconfig):
...
+            )
+        except Exception as exc:
+            log.exception(
+                "Failed to retrieve nodes for ext_pillar", exc_info=exc
+            )
+            pillar_nodes = __utils__['pillar_utils.errors_to_dict']([
+                "Failed to retrieve NodeList: {!s}".format(exc)
+            ])
+        else:
+            log.info("Successfully retrieved nodes for ext_pillar")
```
Comment (gdemonet) [inline salt/_pillar/metalk8s_nodes.py]: I put this for debug purposes, maybe it should be removed or at least toned down to `debug` level

**Review by gdemonet — COMMENTED** [review]
Few things to adjust, will take care of them tomorrow. Thanks a lot @TeddyAndrieux for taking care of this!

**Review by TeddyAndrieux — APPROVED** [review]

**Pull request merged**